### DISCUSSION TITLE
100. MyISAM과 InnoDB에서 B+트리 리프 노드에 데이터를 저장하는 방식의 차이점

### DISCUSSION THREAD

**Discussion by jongmee** [jewelry-box/cs-interview#104] [General]
MyISAM과 InnoDB에서 B+트리 리프 노드에 데이터를 저장하는 방식의 차이점이 뭐죠?

**Comment by linirini** [comment]
특징 | MyISAM | InnoDB
-- | -- | --
리프 노드에 저장되는 값 | 데이터 파일의 물리적 주소 | 실제 데이터 레코드
인덱스 유형 | 비클러스터드 인덱스 | 클러스터드 인덱스
데이터와 인덱스의 관계 | 분리되어 저장 (`.MYD` 파일에 데이터 레코드를, `.MYI` 파일에 인덱스를 물리적으로 분리하여 저장) | 통합되어 저장 (기본 키 인덱스의 리프 노드에 키와 데이터 레코드가 함께 저장된다. 데이터가 인덱스의 일부로 저장되므로 기본 키 인덱스 자체가 데이터의 저장 구조이다.)
데이터 정렬 방식 | 정렬되지 않음 | 기본 키를 기준으로 정렬
추가 인덱스 | 독립적인 비클러스터드 인덱스로 작동 | 보조 인덱스는 클러스터 인덱스의 참조를 포함
